pyproject.toml

```diff
@@ -1,12 +1,173 @@
+[tool.cibuildwheel]
+build = [
+    "cp37-*",
+    "cp38-*",
+    "cp39-*",
+    "cp310-*",
+]
+
+    [tool.cibuildwheel.environment]
+    PIP_PREFER_BINARY = true
+
+    [tool.cibuildwheel.windows]
+    archs = [
+        "AMD64",
+    ]
+
+
+    [tool.cibuildwheel.macos]
+    archs = [
+        "x86_64",
+
+        # Unfortunately we don't have Apple Silicon support yet
+        # TODO: Fix this
+        # Current problem seems to be that installed libfftw3 does not provide arm64
+        # symbols
+        # "universal2",
+        # "arm64",
+    ]
+
+        [tool.cibuildwheel.macos.environment]
+        CC = "/usr/bin/clang"
+        CXX = "/usr/bin/clang++"
+        CFLAGS = "-Wno-implicit-function-declaration -I/usr/local/opt/fftw/include"
+        LDFLAGS = "-Wl,-rpath,/usr/local/opt/fftw/lib -L/usr/local/opt/fftw/lib"
+
+    # On Linux we need to use different package managers depending on image used
+    [tool.cibuildwheel.linux]
+    # Fallback, but we specify versions more explicitly below to achieve maximal
+    # compatibility
+    before-all = "apt update; apt install libfftw3-dev"
+
+    manylinux-x86_64-image = "manylinux_2_24"
+    manylinux-i686-image = "manylinux_2_24"
+    manylinux-aarch64-image = "manylinux_2_24"
+    manylinux-ppc64le-image = "manylinux_2_24"
+    manylinux-s390x-image = "manylinux_2_24"
+    manylinux-pypy_x86_64-image = "manylinux_2_24"
+    manylinux-pypy_i686-image = "manylinux_2_24"
+    manylinux-pypy_aarch64-image = "manylinux_2_24"
+
+    archs = [
+        "x86_64",
+        "i686",
+        "aarch64",
+    ]
+
+    # Select based on manylinux availability
+    # See: https://github.com/pypa/manylinux
+    # See: https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip
+    # See: https://github.com/pypa/manylinux/issues/994#issuecomment-987835591
+    # From 3.10+ we can stop doing the madness below and just PEP 600 manylinux_x_y images
+    # Currently the selection of images is conservative and aims for maximal compatibility
+
+    # The best choice of manylinux is determined by the following:
+    # manylinux version for numpy version for given python version
+
+    # Use manylinux1 for Python3.7, as <3.7.3 does not initially work with manylinux2010
+    [[tool.cibuildwheel.overrides]]
+    select = "cp37-manylinux*"
+
+    manylinux-x86_64-image = "manylinux1"
+    manylinux-i686-image = "manylinux1"
```

Comment:
I think some packages went ahead and also bumped these to manylinux2010, but it is true that this doesn't work for the version of `pip` bundled with Python 3.7. seems fine to leave it at manylinux1 in this Python 3.7 case.

Reply:
I think these are best kept as manylinux1 so it does actually work for all Python 3.7 installations. Support for manylinux1 is disappearing soon (in pypa/manylinux) and once Python3.7 support can be dropped, we can start dropping some of these overrides.

In Python 3.10+ there should be no need for overrides at all.